framework/transport/ieee1905_transport/ieee1905_transport_network.cpp

```diff
@@ -230,18 +174,23 @@ void Ieee1905Transport::handle_interface_status_change(unsigned int if_index, bo
 
     MAPF_DBG("interface " << if_index << " is now " << (is_active ? "active" : "inactive") << ".");
 
-    if (!is_active && network_interfaces_[if_index].fd >= 0) {
-        poller_.Remove(network_interfaces_[if_index].fd);
-        close(network_interfaces_[if_index].fd);
-        network_interfaces_[if_index].fd = -1;
+    if (!is_active && network_interfaces_[if_index].ieee1905_socket >= 0) {
+        poller_.Remove(network_interfaces_[if_index].ieee1905_socket);
+        poller_.Remove(network_interfaces_[if_index].lldp_socket);
+        close(network_interfaces_[if_index].ieee1905_socket);
+        close(network_interfaces_[if_index].lldp_socket);
+        network_interfaces_[if_index].ieee1905_socket = -1;
+        network_interfaces_[if_index].lldp_socket     = -1;
     }
 
-    if (is_active && network_interfaces_[if_index].fd < 0) {
+    if (is_active && network_interfaces_[if_index].ieee1905_socket < 0) {
         if (!open_interface_socket(if_index)) {
             MAPF_ERR("cannot open network interface " << if_index << ".");
         }
-        if (network_interfaces_[if_index].fd >= 0)
-            poller_.Add(network_interfaces_[if_index].fd);
+        if (network_interfaces_[if_index].ieee1905_socket >= 0) {
+            poller_.Add(network_interfaces_[if_index].ieee1905_socket);
+            poller_.Add(network_interfaces_[if_index].lldp_socket);
```

Comment:
Here as well, condition must be duplicated.

Yes, it's true that when ieee1905_socket opening failed you also set lldp_socket to -1, but checking them separately is more defensive.